src/smc-webapp/project_new.cjsx

```diff
@@ -238,8 +243,10 @@ ProjectNewForm = rclass ({name}) ->
         e.preventDefault()
         if @state.filename[@state.filename.length - 1] == '/'
             @create_folder()
-        else
+        else if @state.filename.includes(".")
```

Comment:
One thing I thought about doing was checking if the @state.filename has one of the new_file_button_types extensions. If it does have it, then it would create the file. If it doesn't it would ask if the user wants to create a file with an invalid extension.